airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -56,24 +67,37 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         google_api = GoogleAds(credentials=self.get_credentials(config), customer_id=config["customer_id"])
+        time_zone = self.get_time_zone(google_api)
         incremental_stream_config = dict(
-            api=google_api, conversion_window_days=config["conversion_window_days"], start_date=config["start_date"]
+            api=google_api,
+            conversion_window_days=config["conversion_window_days"],
+            start_date=config["start_date"],
+            time_zone=time_zone
         )
 
-        custom_query_streams = [
-            CustomQuery(custom_query_config=single_query_config, **incremental_stream_config)
-            for single_query_config in config.get("custom_queries", [])
-        ]
-        return [
-            UserLocationReport(**incremental_stream_config),
-            AccountPerformanceReport(**incremental_stream_config),
-            DisplayTopicsPerformanceReport(**incremental_stream_config),
-            DisplayKeywordPerformanceReport(**incremental_stream_config),
-            ShoppingPerformanceReport(**incremental_stream_config),
-            AdGroupAdReport(**incremental_stream_config),
+        streams = [
             AdGroupAds(api=google_api),
             AdGroups(api=google_api),
             Accounts(api=google_api),
             Campaigns(api=google_api),
             ClickView(**incremental_stream_config),
-        ] + custom_query_streams
+        ]
+
+        custom_query_streams = [
+            CustomQuery(custom_query_config=single_query_config, **incremental_stream_config)
+            for single_query_config in config.get("custom_queries", [])
+        ]
+        streams.extend(custom_query_streams)
+
+        is_manager_account = "login_customer_id" in config and config["login_customer_id"].strip()
```

Comment:
I think this is not foolproof. The contract is: 

`customer_id` could be either manager account or non-manager account. If it's non-manager, then `login_customer_id` should be set to the ID of the manager account. So just looking at the config might not be sufficient

Reply:
If I understand correctly, there may be two cases:
1. When [customer_id](https://developers.google.com/google-ads/api/docs/first-call/overview?hl=en) is specified as manager account. In this case, we can get data from all streams.
2. When `customer_id` is specified as a non-manager account and `login_customer_id` is [specified](https://developers.google.com/google-ads/api/docs/concepts/call-structure#cid) as a manager account. In this case, not all streams are available.

Or are you talking about adding a check that the `customer_id` is a manager account, then making available all streams?

Reply:
1. when customer ID is a manager account, we cannot get data from all streams, only the ones available for manager account 
2. when customer ID is not a manager account, can get from all streams

Reply:
Updated. We request account information from Google Analytics and use the value from `customer.manager` field.